### ISSUE TITLE
Keyword-only first argument in instance method not reported

### ISSUE THREAD

**Issue by decorator-factory** [microsoft/pyright#11036]
Tested on Pyright 1.1.406, Python 3.14

```py
class A:
    def meth1(*, x: int = 42) -> None:
        pass

    def meth2(*, x: int) -> None:
        pass

A().meth1()  # no error
A().meth2(x=57)  # no error
```

Both method calls will fail at runtime. Pyright does not report any diagnostics for this file.

`mypy` does report errors both for the method definitions and for the calls (so that e.g. you can keep these functions if you need backwards compatibility, while still keeping type safety when they are used).

**Comment by erictraut** [comment]
Yes, this isn't a check that pyright performs. 

Converting this from a bug report to an enhancement request.
